src/app/documentation-messages.service.ts

```diff
@@ -0,0 +1,69 @@
+import { Injectable } from '@angular/core';
+import { HttpClient, HttpErrorResponse } from '@angular/common/http';
+import { NotificationService, NotificationType } from 'patternfly-ng';
+import { BehaviorSubject, Observable } from 'rxjs';
+import { filter, map } from 'rxjs/operators';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class DocumentationMessagesService {
+
+  private messages: Map<string, string>;
+  private readonly ready = new BehaviorSubject<boolean>(false);
+
+  constructor(
+    private http: HttpClient,
+    private notifications: NotificationService,
+  ) {
+    this.setLocale(navigator.language);
+  }
+
+  setLocale(locale: string = navigator.language) {
+    const languages = navigator.languages.filter((v) => v !== locale);
+    languages.unshift(locale);
+    const locales = [];
+    languages.forEach(lang => {
+      if (lang.indexOf('-') === -1) {
+        locales.push(lang);
+      } else {
+        locales.push(lang);
+        locales.push(lang.split('-')[0]);
+      }
+    });
+    const range = locales.map((v, i) => {
+      if (i > 10) {
+        i = 10;
+      }
+      return `${v};q=${1 - i * 0.1}`;
+    }).join(',');
```

Comment:
I think this section above needs some comments explaining what it's doing and why, or just to be split into some smaller functions so that the function names can serve as documentation. I'm assuming it's because you can have locale names like "en-CA" and "en-US", and you're trying to process both of those into "en"? But beyond that I'm not really sure what's happening.

Reply:
The confusing part is for assigning the q-factor weightings, which are the priorities, to each locale.

I've updated to extract this process into a separate function with proper documentation.